src/mon/LogMonitor.cc

```diff
@@ -374,6 +379,52 @@ void LogMonitor::log_external_close_fds()
   channel_fds.clear();
 }
 
+/// catch external logs up to summary.version
+void LogMonitor::log_external_backlog()
+{
+  if (!external_log_to) {
+    std::string cur_str;
+    int r = mon.store->read_meta("external_log_to", &cur_str);
+    if (r == 0) {
+      external_log_to = atoll(cur_str.c_str());
+      dout(10) << __func__ << " initialized external_log_to = " << external_log_to
+	       << " (recorded log_to position)" << dendl;
+    } else {
+      // pre-quincy, we assumed that anything through summary.version was
+      // logged externally.
+      external_log_to = summary.version;
+      dout(10) << __func__ << " initialized external_log_to = " << external_log_to
+	       << " (summary v " << summary.version << ")" << dendl;
+    }
+  }
+  assert(external_log_to <= get_last_committed());
+  if (external_log_to == get_last_committed()) {
+    return;
+  }
+  if (external_log_to < summary.version) {
+    derr << __func__ << " local logs at " << external_log_to
+	 << ", skipping to " << get_first_committed() << dendl;
+    // FIXME: write marker in each channel log file?
+  }
+  while (external_log_to < summary.version) {
```

Comment:
nit, might want to put `++external_log_to;` closer to the loop condition check. like
```c++
for (; external_log_to < summary.version; ++external_log_to) {
  // ...
}
```